File: Editorial/2574_Left_and_Right_Sum_Differences/C/1_Approach_Prefix_Sum.c
int* leftRightDifference(int* nums, int numsSize, int* returnSize) {
    *returnSize = numsSize;
    int* ans = (int*)malloc(numsSize * sizeof(int));

    int leftSum = 0;
    for (int i = 0; i < numsSize; ++i) {
        ans[i] = leftSum;
        leftSum += nums[i];
    }

    int rightSum = 0;
    for (int i = numsSize - 1; i >= 0; --i) {
        ans[i] = abs(ans[i] - rightSum);
        rightSum += nums[i];
    }

    return ans;
}

File: Editorial/2574_Left_and_Right_Sum_Differences/Go/1_Approach_Prefix_Sum.go
func leftRightDifference(nums []int) []int {
	n := len(nums)
	ans := make([]int, n)

	leftSum := 0
	for i := 0; i < n; i++ {
		ans[i] = leftSum
		leftSum += nums[i]
	}

	rightSum := 0
	for i := n - 1; i >= 0; i-- {
		ans[i] = abs(ans[i] - rightSum)
		rightSum += nums[i]
	}

	return ans
}

func abs(x int) int {
	if x < 0 {
		return -x
	}
	return x
}
